corporate/lib/stripe.py

```diff
@@ -1096,25 +1121,58 @@ def process_initial_upgrade(
                             _("Please add a credit card before starting your free trial."),
                         )
 
+            event_time = billing_cycle_anchor
+            if should_schedule_upgrade_for_legacy_remote_server:  # nocoverage
+                assert remote_server_legacy_plan is not None
+                if charge_automatically:
+                    remote_server_legacy_plan.charge_automatically = True
+                    # Ensure customers not paying via invoice have a default payment method set.
+                    stripe_customer = stripe_get_customer(customer.stripe_customer_id)
+                    if not stripe_customer_has_credit_card_as_default_payment_method(
+                        stripe_customer
+                    ):
+                        raise BillingError(
+                            "no payment method",
+                            _("Please add a credit card to schedule upgrade."),
+                        )
+
+                # Settings status > CustomerPLan.LIVE_STATUS_THRESHOLD makes sure we don't have
+                # to worry about this plan being used for any other purpose.
+                # NOTE: This is the 2nd plan for the customer.
+                plan_params["status"] = CustomerPlan.NEVER_STARTED
+                event_time = timezone_now().replace(microsecond=0)
```

Comment:
We want `event_time` to be reflected correctly as `now` in `LicenseLedger` and audit log created below but `billing_cycle_anchor` is set at `legacy_plan.end_date` since we want the second plan to start when the legacy_plan ends.

Reply:
I understand why now should be what we put in the audit log. I guess as noted in https://github.com/zulip/zulip/pull/28017#discussion_r1414367096, these license ledger entries are a bit of a weird hack, maybe now is the best `event_time` for them but it's all a bit weird.